pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -165,12 +167,8 @@ func getMachineHealthCheckListOptions() *client.ListOptions {
 	}
 }
 
-func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconcile.Result, error) {
+func remediate(r *ReconcileMachineHealthCheck, hc *healthcheckingv1alpha1.MachineHealthCheck, machine *mapiv1.Machine) (reconcile.Result, error) {
```

Comment:
Do we need the whole `MachineHealthCheck` here? If all we need is the remediation strategy, can we just pass that?